src/modules/esl-media/core/esl-media.ts

```diff
@@ -159,6 +163,15 @@ export class ESLMedia extends ESLBaseElement {
       case 'media-type':
         this.deferredReinitialize();
         break;
+      case 'lazy':
+        if (this.lazy === 'manual') {
+          this.deferredReinitialize();
+        } else if (this.lazy === 'auto') {
+          this.attachViewportConstraint();
+        } else {
+          this.detachViewportConstraint();
+        }
```

Comment:
In order to reduce the complexity of the method, can we put this code in a separate method, for example, `onLazyAttributeChange()`?